tests/tests/htlc_tests.cpp

```diff
@@ -259,6 +260,72 @@ try {
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      graphene::chain::proposal_create_operation prop1;
+      prop1.fee_paying_account = alice_id;
+      prop1.expiration_time = db.head_block_time() + 100;
+      prop1.proposed_ops.emplace_back(create_operation);
+      trx.operations.push_back(prop1);
+      sign(trx, alice_private_key);
+      PUSH_TX(db, trx, ~0);
+      trx.clear();
+   }
+   // Alice attempts to put a proposal out with a memo (which is not valid)
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.extensions.value.memo = memo_data();
+      create_operation.extensions.value.memo->from = alice_public_key;
+      create_operation.extensions.value.memo->to = bob_public_key;
+      create_operation.extensions.value.memo->set_message(
+            alice_private_key, bob_public_key, "Dear Bob,\n\nMoney!\n\nLove, Alice");      
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      graphene::chain::proposal_create_operation prop1;
+      prop1.fee_paying_account = alice_id;
+      prop1.expiration_time = db.head_block_time() + 100;
+      prop1.proposed_ops.emplace_back(create_operation);
+      trx.operations.push_back(prop1);
+      sign(trx, alice_private_key);
+      GRAPHENE_CHECK_THROW(PUSH_TX(db, trx, ~0), fc::exception);
+      trx.clear();
+   }   
+   // Alice attempts to put a contract on the blockchain that is correct but has a preimage size of 0
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.extensions.value.memo = memo_data();
```

Comment:
And memo should be empty, so the exception would be due to the `HASH160` algorithm but not the memo.